src/rsa/signing.rs

```diff
@@ -176,10 +213,18 @@ impl RSAKeyPair {
                     return Err(error::Unspecified);
                 }
 
+                // [NIST SP-800-56B rev. 1]] 6.4.1.4.3 - Step 6
+                //
+                // TODO 6.a: d < LCM(p – 1, q – 1)
+                // TODO 6.b: 1 == (d · e) mod LCM(p – 1, q – 1)
+                //
+                // [NIST SP-800-56B rev. 1]] 6.4.1.4.3 - Step 6.a, partial
                 if !(d.bit_length() > half_n_bits) {
```

Comment:
The bit length of `2**n_bits/2` should be `half_n_bits + 1`, I think, so I think this is correct as-is: this gives us `d >= 2**(n_bits/2)` and knowing `d` is odd makes that `d > 2**(n_bits/2)`. But I'll add a comment.